ui/packages/catalog-ui-search/src/main/webapp/component/search-form/search-form.view.js

```diff
@@ -55,8 +56,38 @@
                 this.options.queryModel.set('type', 'text');
                 user.getQuerySettings().set('type', 'text');
                 break;
+            case 'newResult':
+                oldType = this.options.queryModel.get('type');
+                this.options.queryModel.set({
+                    type: 'newResult',
+                    title: "",
+                    modelId: "",
+                    accessGroups: [],
+                    accessIndividuals: [],
+                    descriptors: [],
+                    description: ""
+                });
+                if (oldType  === 'newResult') {
+                    this.options.queryModel.trigger('change:type');
+                }
+                break;              
+            case 'result':
+                oldType = this.options.queryModel.get('type');
+                this.options.queryModel.set({
+                    type: 'result',
+                    title: this.model.get('name'),
+                    modelId: this.model.get('id'),
+                    accessGroups: this.model.get('accessGroups'),
+                    accessIndividuals: this.model.get('accessIndividuals'),
+                    descriptors: this.model.get('descriptors'),
+                    description: this.model.get('description')
+                });
+                if (oldType  === 'result') {
+                    this.options.queryModel.trigger('change:type');
+                }
+                break;              
             case 'custom':
-                var oldType = this.options.queryModel.get('type');
+                oldType = this.options.queryModel.get('type');
                 this.options.queryModel.set({
```

Comment:
Instead of setting an object here, change it to a new instantiation of SearchForm model so we don't keep around old data. See comment above.